src/utils.lisp

```diff
@@ -97,19 +96,31 @@ accumulated, in the order."
   `(let ((it ,condition))
      (when it ,@body)))
 
-(defun integer-to-string (x)
+(defun integer-to-string (x &optional (radix (or *print-base* 10)) plusp)
```

Comment:
I think the heart of this (with the order-of-sign fix) can/should port up into a print-number function perhaps, then, to preserve the `*print-radix*`/`*print-base*` logic between `print` & `format`?

I didn't even realize the `#x-ff` case, totally missed that. I should probably add a test case for that.